clients/src/main/java/org/apache/kafka/common/requests/FetchResponse.java

```diff
@@ -99,28 +99,24 @@ public Errors error() {
     }
 
     public LinkedHashMap<TopicPartition, FetchResponseData.PartitionData> responseData(Map<Uuid, String> topicNames, short version) {
-        if (responseData == null) {
-            synchronized (this) {
-                if (responseData == null) {
-                    // Assigning the lazy-initialized `responseData` in the last step
-                    // to avoid other threads accessing a half-initialized object.
-                    final LinkedHashMap<TopicPartition, FetchResponseData.PartitionData> responseDataTmp =
-                            new LinkedHashMap<>();
-                    data.responses().forEach(topicResponse -> {
-                        String name;
-                        if (version < 13) {
-                            name = topicResponse.topic();
-                        } else {
-                            name = topicNames.get(topicResponse.topicId());
-                        }
-                        if (name != null) {
-                            topicResponse.partitions().forEach(partition ->
-                                responseDataTmp.put(new TopicPartition(name, partition.partitionIndex()), partition));
-                        }
-                    });
-                    responseData = responseDataTmp;
+        synchronized (this) {
+            // Assigning the lazy-initialized `responseData` in the last step
+            // to avoid other threads accessing a half-initialized object.
+            final LinkedHashMap<TopicPartition, FetchResponseData.PartitionData> responseDataTmp =
+                    new LinkedHashMap<>();
+            data.responses().forEach(topicResponse -> {
+                String name;
+                if (version < 13) {
+                    name = topicResponse.topic();
+                } else {
+                    name = topicNames.get(topicResponse.topicId());
                 }
-            }
+                if (name != null) {
+                    topicResponse.partitions().forEach(partition ->
+                        responseDataTmp.put(new TopicPartition(name, partition.partitionIndex()), partition));
+                }
+            });
+            responseData = responseDataTmp;
```

Comment:
This is what I do concern! The `responseData` could be different if the input gets changed.

Reply:
thanks for the review!
yes, the returned data should be calculated on the fly based on the input topic names